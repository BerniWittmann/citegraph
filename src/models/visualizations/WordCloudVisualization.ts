import Visualization, { VisualizationBaseFields } from './Visualization'

export interface WordCloudVisualizationParameters {
  pruneLeastCitedPercentage: number
}

export interface WordCloudDataPoint {
  word: string
  weight: number
}
export interface WordCloudData {
  data: Array<WordCloudDataPoint>
}

export interface WordCloudVisualizationFields {
  data?: WordCloudData
}

export default class WordCloudVisualization extends Visualization implements WordCloudVisualizationFields {
  static key: string = 'wordcloud'
  key: string = 'wordcloud'
  static title: string = 'visualizations.information.word_cloud.title'
  static description: string = 'visualizations.information.word_cloud.description'
  static longDescription: string = 'visualizations.information.word_cloud.long_description'
  static imageUrl: string = 'visualization_images/word_cloud.jpg'
  parameters: WordCloudVisualizationParameters
  data?: WordCloudData

  constructor ({ id, name, progress, data }: VisualizationBaseFields & WordCloudVisualizationFields) {
    super({ id, name, progress })
    this.parameters = {
      pruneLeastCitedPercentage: 5
    }
    this.data = data
  }
}
