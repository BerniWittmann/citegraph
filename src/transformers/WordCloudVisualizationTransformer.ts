/* eslint-disable camelcase */
import Transformer from './Transformer'
import WordCloudVisualization from '@/models/visualizations/WordCloudVisualization'
import { VisualizationBaseFieldsDTO } from '@/models/visualizations/Visualization'

export interface WordCloudDataPointDTO {
  word: string
  weight: number
}

export interface WordCloudVisualizationDTO extends VisualizationBaseFieldsDTO {
  data?: Array<WordCloudDataPointDTO>
}

export default class WordCloudVisualizationTransformer extends Transformer {
  static fetch (item: WordCloudVisualizationDTO): WordCloudVisualization {
    return new WordCloudVisualization({
      id: item.id,
      name: item.name,
      progress: item.progress,
      timePeriods: item.time_periods,
      data: item.data ? {
        data: item.data
      } : undefined
    })
  }

  static send (item: WordCloudVisualization): WordCloudVisualizationDTO {
    return {
      id: item.id,
      name: item.name,
      key: item.key,
      progress: item.progress,
      time_periods: item.timePeriods,
      data: item.data ? item.data.data : undefined
    }
  }
}
