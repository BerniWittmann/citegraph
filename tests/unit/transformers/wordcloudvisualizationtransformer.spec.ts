import WordCloudVisualizationTransformer from '@/transformers/WordCloudVisualizationTransformer'
import WordCloudVisualization from '@/models/visualizations/WordCloudVisualization'

describe('transformers/WordCloudVisualizationTransformer', () => {
  describe('can send a WordCloudVisualization', () => {
    it('sends an WordCloudVisualization', () => {
      expect(WordCloudVisualizationTransformer.send(new WordCloudVisualization({ id: '1', name: 'First', progress: 0.1 }))).toMatchSnapshot()
    })

    it('sends the data', () => {
      expect(WordCloudVisualizationTransformer.send(new WordCloudVisualization({ id: '1', name: 'First', progress: 0.1, data: { data: [] } }))).toMatchSnapshot()
    })
  })

  describe('can fetch a WordCloudVisualization', () => {
    it('fetches a WordCloudVisualization', () => {
      expect(WordCloudVisualizationTransformer.fetch({ id: '1', name: 'First', progress: 0.1, key: 'wordcloud' })).toMatchSnapshot()
    })

    it('fetches the data', () => {
      expect(WordCloudVisualizationTransformer.fetch({ id: '1', name: 'First', progress: 0.1, key: 'wordcloud', data: [] })).toMatchSnapshot()
    })
  })
})
