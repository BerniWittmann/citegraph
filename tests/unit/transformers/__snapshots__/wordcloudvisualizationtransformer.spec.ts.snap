// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformers/WordCloudVisualizationTransformer can fetch a WordCloudVisualization fetches a WordCloudVisualization 1`] = `
WordCloudVisualization {
  "data": undefined,
  "id": "1",
  "key": "wordcloud",
  "name": "First",
  "parameters": Object {
    "pruneLeastCitedPercentage": 5,
  },
  "progress": 0.1,
  "timePeriods": Array [],
}
`;

exports[`transformers/WordCloudVisualizationTransformer can fetch a WordCloudVisualization fetches the data 1`] = `
WordCloudVisualization {
  "data": Object {
    "data": Array [],
  },
  "id": "1",
  "key": "wordcloud",
  "name": "First",
  "parameters": Object {
    "pruneLeastCitedPercentage": 5,
  },
  "progress": 0.1,
  "timePeriods": Array [],
}
`;

exports[`transformers/WordCloudVisualizationTransformer can send a WordCloudVisualization sends an WordCloudVisualization 1`] = `
Object {
  "data": undefined,
  "id": "1",
  "key": "wordcloud",
  "name": "First",
  "progress": 0.1,
  "time_periods": Array [],
}
`;

exports[`transformers/WordCloudVisualizationTransformer can send a WordCloudVisualization sends the data 1`] = `
Object {
  "data": Array [],
  "id": "1",
  "key": "wordcloud",
  "name": "First",
  "progress": 0.1,
  "time_periods": Array [],
}
`;
